### DISCUSSION TITLE
invoke changed?

### DISCUSSION THREAD

**Discussion by bezio** [deployphp/deployer#2581] [General]
Just a question if invoke has accidentally lost some functionality from 6.x to 7.0beta?
On 6.x the invoke() respects selects/on-calls on the invoked tasks - in 7.0beta the invoke'd task is simply executed without this check....

Is that intentionall?

**Comment by antonmedv** [comment]
Yes, now invoke is only for calling functions.

**Reply by bezio** [reply]
Hm. Ok, that's sad.
We used this as a filter so we only need to define rules on the task and invoked them (getting filtered).
Now we need to prepare other ways...

Thanks for your fast reply!
